libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -399,15 +436,131 @@ void AP_Airspeed::read(void)
+        // If there are two sensors then the first check will be true for both, however
+        // the second check will be true only for the sensor which really failed
+
+        // TODO: decide on minimum speed for checking (now hard set to 5.0)
+        for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+            if (state[i].blend_healthy &&
+                    ekf_sensor_airspeed_diff[i] > airspeed_avg_limit && state[i].airspeed > 5.0) {
+                gcs().send_text(MAV_SEVERITY_INFO, "Airspeed[%u] sensor blending unhealthy", i);
+                state[i].blend_healthy = false;
+            }
+        }
+
+        // TODO: Check what is really needed and where is used. Maybe something is missed.
+        blend_state.airspeed = 0.0;
+        blend_state.raw_airspeed = 0.0;
+        blend_state.temperature = 0.0;
+        blend_state.last_pressure = 0.0;
+        blend_state.corrected_pressure = 0.0;
+        blend_state.EAS2TAS = 0.0;
+        blend_state.ratio = 0.0;
+        blend_state.offset = 0.0;
+
+        healthy_sens_num = 0;
+
+        for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+            if (!state[i].blend_healthy) {
+                continue;
+            }
+            
+            blend_state.airspeed += state[i].airspeed;
+            blend_state.raw_airspeed += state[i].raw_airspeed;
+
+            // TODO: Check for get_temperature result and correction to number of sensors
+            float temperature;
+            get_temperature(i, temperature);
+            blend_state.temperature += temperature;
+
+            blend_state.last_pressure += state[i].last_pressure;
+            blend_state.corrected_pressure += state[i].corrected_pressure;
+            blend_state.EAS2TAS += state[i].EAS2TAS;
+            
+            blend_state.ratio += param[i].ratio;
+            blend_state.offset += param[i].offset;
+
+            healthy_sens_num++;
+        }
+        
+        if (healthy_sens_num > 0) {
+            float div2mul_factor = 1.0 / healthy_sens_num;
+            blend_state.airspeed *= div2mul_factor;
+            blend_state.raw_airspeed *= div2mul_factor;
+            blend_state.temperature *= div2mul_factor;
+            blend_state.last_pressure *= div2mul_factor;
+            blend_state.corrected_pressure *= div2mul_factor;
+            blend_state.EAS2TAS *= div2mul_factor;
+            blend_state.ratio *= div2mul_factor;
+            blend_state.offset *= div2mul_factor;
+            
+            blend_state.blend_active = true;
+        } else {
+            process_no_blend();
+        }
+    } else
+    {
+        process_no_blend();
+    }
+    
+    // Check if any unhealthy sensors came back into range
+    // TODO: decide if this is really needed
+    for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+        if (!state[i].blend_healthy && ekf_sensor_airspeed_diff[i] < 5.0 &&
```

Comment:
condition isn't the same as previous, here you just check ekf_sensor_airspeed_diff[i] < 5.0

Reply:
Yes, and they can't be the same as in this case the avg can be 0 or whatever. Needs to be absolute.